src/checkFirebaseSDKVersion.ts

```diff
@@ -0,0 +1,91 @@
+"use strict";
+
+import * as _ from "lodash";
+import * as clc from "cli-color";
+import * as path from "path";
+import * as semver from "semver";
+import * as spawn from "cross-spawn";
+import * as utils from "./utils";
+import * as logger from "./logger";
+
+interface NpmListResult {
+  name: string;
+  dependencies: {
+    "firebase-functions": {
+      version: string;
+      from: string;
+      resolved: string;
+    };
+  };
+}
+
+interface NpmShowResult {
+  "dist-tags": {
+    latest: string;
+  };
+}
+
+export async function getFunctionsSDKVersion(sourceDir: string): Promise<string | void> {
+  try {
+    const child = spawn.sync("npm", ["list", "firebase-functions", "--json=true"], {
+      cwd: sourceDir,
+      encoding: "utf8",
+    });
+    if (child.error) {
+      logger.debug(child.error.stack);
+      return;
+    }
+    const output: NpmListResult = JSON.parse(child.stdout);
+    return _.get(output, ["dependencies", "firebase-functions", "version"]);
+  } catch (e) {
+    logger.debug("getFunctionsSDKVersion encountered error:", e);
+    return;
+  }
+}
+
+export async function checkSDKVersion(options: any): Promise<void> {
```

Comment:
With the system calls being `spawn.sync`, I don't think this has to be `async` either